tests/test_cliCmdQueryDataIds.py

```diff
@@ -112,7 +112,7 @@ def testNoResultsEasy(self):
             where="instrument='Cam1' AND skymap='SkyMap1' AND visit=1 AND tract=1",
         )
         self.assertIsNone(res, msg)
-        self.assertIn("yields no results when applied to", msg)
```

Comment:
This message comes from daf_relation and new query system doesn't report any failure reason.

Reply:
:+1:.  Explaining no results is something the new query system does not do as well yet, though the old system was actually pretty hit or miss and the tests kind of disguised that.  The old system walked back through the tree of joins and WHERE constraints, executing LIMIT 1 queries to see where the result set is first empty, but it never tried applying those joins and WHERE constraints in a different order.